avocado/utils/network/hosts.py

```diff
@@ -124,6 +124,9 @@ def __init__(self, host, username, port=22, key=None, password=None):
         self.password = password
         self.remote_session = self._connect()
 
+    def __exit__(self):
+        self.remote_session.quit()
```

Comment:
This is a nice idea. But since self._connect can raise exceptions, self.remove_session can be empty.

So I would add a basic check here:

```python
if self.remote_sesion:
    self.remote_session.quit()
```

Reply:
Good point. Thanks for the quick review!